src/run/handlers/use-cache-handler.ts

```diff
@@ -0,0 +1,270 @@
+import { Buffer } from 'node:buffer'
+
+import { LRUCache } from 'lru-cache'
+import type {
+  CacheEntry,
+  // only supporting latest variant (https://github.com/vercel/next.js/pull/76687)
+  // first released in v15.3.0-canary.13
+  CacheHandlerV2 as CacheHandler,
+} from 'next-next/dist/server/lib/cache-handlers/types.js'
+
+import { getLogger } from './request-context.cjs'
+import {
+  getMostRecentTagRevalidationTimestamp,
+  isAnyTagStale,
+  markTagsAsStaleAndPurgeEdgeCache,
+} from './tags-handler.cjs'
+import { getTracer } from './tracer.cjs'
+
+// copied from default implementation
+// packages/next/src/server/lib/cache-handlers/default.ts
+type PrivateCacheEntry = {
+  entry: CacheEntry
+
+  // For the default cache we store errored cache
+  // entries and allow them to be used up to 3 times
+  // after that we want to dispose it and try for fresh
+
+  // If an entry is errored we return no entry
+  // three times so that we retry hitting origin (MISS)
+  // and then if it still fails to set after the third we
+  // return the errored content and use expiration of
+  // Math.min(30, entry.expiration)
+
+  isErrored: boolean // pieh: this doesn't seem to be actually used in the default implementation
+  errorRetryCount: number // pieh: this doesn't seem to be actually used in the default implementation
```

Comment:
Those are only ever set, but never actually read, so possibly those should be removed. This interface is copied as-is from Next.js default implementation, so I did want to include this intentionally here for awareness

Reply:
This was removed in https://github.com/opennextjs/opennextjs-netlify/pull/2862/commits/8a23ceaaccc81e993c207ba07c2159eeb95a6baf